### PR TITLE
Fix/suite replace transaction action

### LINKED ISSUE #7856: TX detail modal doesn't hot reload changes
**Describe the bug**
When the user eg. bumps a fee of their tx and keep their modal open, the info is not updated in real time.

**Steps to reproduce:**
1. Send a tx with `low` fee
2. Bump the fee to `high`
3. Open the tx detail and wait
4. Look up the tx on btc explorer (eg. `https://btc5.trezor.io/`)

**Info:**
 - Suite version: desktop 23.2.1 (bcfbc9ffb59b51f6c7ecb58d529769ca0db2f7b1)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/23.2.1 Chrome/104.0.5112.124 Electron/20.3.5 Safari/537.36
 - OS: MacIntel
 - Screen: 2560x1440
 - Device: model T 2.5.3 Bitcoin-only (revision 2f03ace311584988d5aeab58fd1acf24ef54711a)
 - Transport: BridgeTransport 2.0.32

**Expected result:**
The tx data is updated in real time, as the fee is bumped or the tx is confirmed

**Actual result:**
The tx data is not updated unless the tx is confirmed

**Screenshots:**
image

### LINKED ISSUE #8724: Fee rate in transaction detail not updated after transaction bump
**Describe the bug**
Fee rate not updated after transaction bump in transaction detail.
Fee changes but not fee rates, easy to compare with blockbook that it does not match

**Steps to reproduce:**
1. Send transaction and check transaction detail
2. Bump or finalize transaction and check transaction detail
3. Fee rate never changes even after multiple bumps :)

**Info:**
 - Suite version: desktop 23.6.1 (bddd516378beadfcff94523e7c94834ba14f147f)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/23.6.1 Chrome/112.0.5615.183 Electron/24.3.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: model T 2.6.1 Universal (revision 089ff768229f543c89927ff34c4d45d830a2442f)
 - Transport: BridgeTransport 2.0.33

**Screenshots:**
Screenshot 2023-06-19 at 12 33 42

Screenshot 2023-06-19 at 12 32 18

image

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description
Minor adjustments here and there + fixing issues related to RBF for both bitcoin and ethereum as prerequisite of https://github.com/trezor/trezor-suite/pull/9040

## Explanation

- Timestamp of pending transaction was not always visible ([mentioned here](https://github.com/trezor/trezor-suite/issues/8719#issuecomment-1602364030)) because of invalid condition in the `TimestampLink` component. `blockHeight` could be <= 0 (usually is set to -1). IMHO blockHeight should not be taken in consideration there

- `@trezor/connect` createPendingTx didn't use `sequence` in inputs so RBF "fake pending tx" (prepending) was always shown as final

- fixed/refactored `replaceTransactionThunk` action (update data in tx detail modal)
  - for bitcoin-like use signedTransaction from `@trezor/connect` same as in addFakePendingTxThunk - this creates exact same transaction as received later from the backend
  - for ethereum-like update `ethereumSpecific` and use `getRbfParams`
  
- skip creation on "fake pending tx" if it was a replacement tx - unnecessary step, tx will be ignored anyway (already updated by replaceTransactionThunk)

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/8724
Resolve https://github.com/trezor/trezor-suite/issues/7856

### CODE REVIEW

**Review by tomasklim — APPROVED** [review]
ETH stuff

File: suite-common/wallet-core/src/transactions/transactionsReducer.ts [hunk 1]
```diff
@@ -125,6 +125,9 @@ export const prepareTransactionsReducer = createReducerWithExtraDeps(
                             ((existingTx.blockHeight ?? 0) <= 0 &&
                                 (transaction.blockHeight ?? 0) > 0) ||
                             (!existingTx.blockTime && transaction.blockTime) ||
+                            (existingTx.blockTime &&
+                                transaction.blockTime &&
+                                existingTx.blockTime < transaction.blockTime) ||
```
Comment (marekrjpolak) [inline suite-common/wallet-core/src/transactions/transactionsReducer.ts]: Same as this?
```suggestion
                            (transaction.blockTime &&
                                (existingTx.blockTime ?? 0) < transaction.blockTime) ||
```

**Review by marekrjpolak — APPROVED** [review]
Apart from the one and only nitpick, it's good.

File: suite-common/wallet-core/src/transactions/transactionsReducer.ts (hunk 1 above)
Comment (szymonlesisz) [reply suite-common/wallet-core/src/transactions/transactionsReducer.ts]: [b418dbd](https://github.com/trezor/trezor-suite/pull/9058/commits/b418dbdf1b87849217603241e1b53ab0fe026abf)

**Pull request merged**